File: SDKs/amazonaws.com/route53resolver/2018-04-01/go/pkg/models/operations/createfirewallrule.go
package operations

import (
"openapi/pkg/models/shared")


type CreateFirewallRuleXAmzTargetEnum string

const (
    CreateFirewallRuleXAmzTargetEnumRoute53ResolverCreateFirewallRule CreateFirewallRuleXAmzTargetEnum = "Route53Resolver.CreateFirewallRule"
)


type CreateFirewallRuleHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    XAmzTarget CreateFirewallRuleXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
    
}

type CreateFirewallRuleRequest struct {
    Headers CreateFirewallRuleHeaders 
    Request shared.CreateFirewallRuleRequest `request:"mediaType=application/json"`
    
}

type CreateFirewallRuleResponse struct {
    AccessDeniedException *interface{} 
    ContentType string 
    CreateFirewallRuleResponse *shared.CreateFirewallRuleResponse 
    InternalServiceErrorException *interface{} 
    LimitExceededException *interface{} 
    ResourceNotFoundException *interface{} 
    StatusCode int64 
    ThrottlingException *interface{} 
    ValidationException *interface{} 
    
}

File: SDKs/amazonaws.com/route53resolver/2018-04-01/typescript/src/sdk/models/operations/createfirewallrule.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";

export enum CreateFirewallRuleXAmzTargetEnum {
    Route53ResolverCreateFirewallRule = "Route53Resolver.CreateFirewallRule"
}


export class CreateFirewallRuleHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: CreateFirewallRuleXAmzTargetEnum;
}


export class CreateFirewallRuleRequest extends SpeakeasyBase {
  @Metadata()
  headers: CreateFirewallRuleHeaders;

  @Metadata({ data: "request, media_type=application/json" })
  request: shared.CreateFirewallRuleRequest;
}


export class CreateFirewallRuleResponse extends SpeakeasyBase {
  @Metadata()
  accessDeniedException?: any;

  @Metadata()
  contentType: string;

  @Metadata()
  createFirewallRuleResponse?: shared.CreateFirewallRuleResponse;

  @Metadata()
  internalServiceErrorException?: any;

  @Metadata()
  limitExceededException?: any;

  @Metadata()
  resourceNotFoundException?: any;

  @Metadata()
  statusCode: number;

  @Metadata()
  throttlingException?: any;

  @Metadata()
  validationException?: any;
}
